src/main/java/firsttask/taskmanager/Security/JWTSecurity/JwtUtil.java

```diff
@@ -7,16 +7,30 @@
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Service;
 
-import java.util.Date;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 import java.util.function.Function;
 
 @Service
 public class JwtUtil {
 
     private final String SECRET_KEY = "secret";// this key is used to encrypt the jwt it should be stored in a save file and encrypted and decrypted when needed
+    private static List<String> blackListTokens= new ArrayList<>();
+    private static List<String> loggedInUserTokens= new ArrayList<>();
```

Comment:
You got the idea we need to save tokens somewhere, and that's great, but I have some questions:

1) what's happened if more than one user logged in?
2) what's happened if the server went down, and we got it up again? will we lose the tokens?

Reply:
1- a problem will happen that if one user press logoutall this will logout all of them at once, I'll change it by using map mapping the username with user token and adding just the logged out user
2- I wanted to ask you about it, there is a way that i can store the list of the blacklist tokens inside the database and every for every time check for them and wait for its expiration date so I can delete it from the database( i am keeping it in the database until it expires so that i make sure that user cant use it again)

Reply:
I think saving them in the database will be a good solution.

But another thing came to my mind now, saving the black list of tokens will lead to a huge amount of data in this table, maybe we can do something better.